sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java

```diff
@@ -1119,7 +1185,7 @@ public BigtableWriteException(KV<ByteString, Iterable<Mutation>> record, Throwab
 
   static void validateTableExists(BigtableConfig config, PipelineOptions options) {
     if (config.getValidate()) {
-      String tableId = config.getTableId();
+      String tableId = config.getTableId().get();
```

Comment:
Does it make sense to skip validation if any of the value providers are inaccessible?

Otherwise users will always need to write withTable(myValueProvider).withoutValidation();

Reply:
As per documents, this should be called right before pipeline run, should not be providers value available at that point?

Reply:
Value providers are only useful during pipeline execution and add no value during pipeline construction. If you know the value during pipeline construction why would you need a value provider?

What documents do you refer to so they can be updated/fixed?

Reply:
I was referring to PTransform#validate JavaDoc.